EDDiscovery/UserControls/UserControlScanGrid.cs

```diff
@@ -195,22 +195,53 @@ void DrawSystem(HistoryEntry he, bool force)
+                    if (sn.ScanData.HasBelts || sn.ScanData.HasRings)
                     {
-                        if (sn.ScanData.Rings.Count() > 1)
+                        if (sn.ScanData.Rings[0].Name.EndsWith("Belt"))
                         {
-                            bdDetails.Append(string.Format("Has {0} rings: ".Tx(this, "Rings"), sn.ScanData.Rings.Count()));
+                            if (sn.ScanData.Rings.Count() > 1)
+                            {
+                                bdDetails.Append(string.Format("Has {0} belts: ".Tx(this, "Has {0} belts"),
+                                                               sn.ScanData.Rings.Count()));
+                            }
+                            else
+                            {
+                                bdDetails.Append("Has 1 belt: ".Tx(this, "Has 1 belt"));
+                            }
+
+                            foreach (var r in sn.ScanData.Rings)
+                            {
+                                var BeltName = r.Name;
+                                bdDetails.Append(JournalScan.StarPlanetRing
+                                                            .DisplayStringFromRingClass(r
+                                                                                            .RingClass) + " ");
+                                bdDetails.Append((r.InnerRad / JournalScan.oneLS_m).ToString("N2") +
+                                                 "ls to " + (r.OuterRad / JournalScan.oneLS_m)
+                                                 .ToString("N2") + "ls. ");
+                            }
                         }
-                        else
-                        {
-                            bdDetails.Append("Has 1 ring: ".Tx(this,"Ring"));
-                        }
-
-                        for (int i = 0; i < sn.ScanData.Rings.Length; i++)
+                        else if (sn.ScanData.Rings[0].Name.EndsWith("Ring"))
                         {
-                            string RingName = sn.ScanData.Rings[i].Name;
-                            bdDetails.Append(JournalScan.StarPlanetRing.DisplayStringFromRingClass(sn.ScanData.Rings[i].RingClass) + " ");
-                            bdDetails.Append((sn.ScanData.Rings[i].InnerRad / JournalScan.oneLS_m).ToString("N2") + "ls to " + (sn.ScanData.Rings[i].OuterRad / JournalScan.oneLS_m).ToString("N2") + "ls. ");
+                            if (sn.ScanData.Rings.Count() > 1)
+                            {
+                                bdDetails.Append(string.Format("Has {0} rings: ".Tx(this, "Rings"),
+                                                               sn.ScanData.Rings.Count()));
+                            }
+                            else
+                            {
+                                bdDetails.Append("Has 1 ring: ".Tx(this, "Ring"));
+                            }
+
+                            foreach (var r in sn.ScanData.Rings)
```

Comment:
Should this perhaps be outside the `if (sn.ScanData.Rings[0].Name.EndsWith("{Belt,Ring}"))` - it is identical to foreach block in the Belts block.

Reply:
I have implemented some different solution, that one seemed simple enough. Anyway, I'll wait for the next updates to rework the whole idea. Thanks!